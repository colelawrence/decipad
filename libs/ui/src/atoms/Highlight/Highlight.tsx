/* eslint decipad/css-prop-named-variable: 0 */
import { useThemeFromStore } from '@decipad/react-contexts';
import { Swatch, swatchesThemed } from '@decipad/ui';
import { css } from '@emotion/react';
import { ReactNode } from 'react';
import {
  normalOpacity,
  strongOpacity,
  transparency,
  weakOpacity,
} from '../../primitives';
import { highlight } from '../../styles';

const styles = (baseSwatches: Swatch) => {
  return css(highlight.highlightStyles, {
    borderRadius: '0.9em 0.3em',
    margin: '0 -0.45em',
    padding: '0.15em 0.45em',
    background: 'transparent',
    boxDecorationBreak: 'clone',
    webkitBoxDecorationBreak: 'clone',
    backgroundImage: `linear-gradient(
    to right,
    ${transparency(baseSwatches.Sun, weakOpacity).rgba},
    ${transparency(baseSwatches.Sun, strongOpacity).rgba} 5%,
    ${transparency(baseSwatches.Sun, normalOpacity).rgba}
  )`,
  });
};

interface HighlightProps {
  readonly children: ReactNode;
}
export const Highlight = ({
  children,
}: HighlightProps): ReturnType<React.FC> => {
  const [darkTheme] = useThemeFromStore();
  const baseSwatches = swatchesThemed(darkTheme);

  return <span css={styles(baseSwatches)}>{children}</span>;
};
