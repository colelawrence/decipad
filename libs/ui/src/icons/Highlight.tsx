import type { SVGProps } from 'react';

export const Highlight = (props: SVGProps<SVGSVGElement>) => {
  return (
    <svg
      xmlns="http://www.w3.org/2000/svg"
      fill="none"
      viewBox="0 0 16 16"
      role="img"
      {...props}
    >
      <title>Highlight</title>
      <path
        stroke="currentColor"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth={1.235}
        d="M5 3v2.059c.686 1.235 1.647 4.035 1.647 5.353h3.706C10.353 9.176 12 5.059 12 5.059V3M7.059 13.294v-2.882H9.94v1.647l-2.882 1.235M5 5.059h7"
      />
    </svg>
  );
};
