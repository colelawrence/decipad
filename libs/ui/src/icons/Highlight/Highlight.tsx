import { FC } from 'react';
import { cssVar, yellow500 } from '../../primitives';

export const Highlight = (): ReturnType<FC> => (
  <svg viewBox="0 0 16 16" fill="none" xmlns="http://www.w3.org/2000/svg">
    <title>highlight</title>
    <path
      d="M9.94119 10.4117H7.05884V13.2941L9.94119 12.4706V10.4117Z"
      fill={yellow500.rgb}
    />
    <path
      d="M5 3V5.05882C5.68627 6.29412 6.64706 9.09412 6.64706 10.4118H10.3529C10.3529 9.17647 12 5.05882 12 5.05882V3"
      stroke={cssVar('currentTextColor')}
      strokeWidth="1.23529"
      strokeLinecap="round"
      strokeLinejoin="round"
    />
    <path
      d="M7.05884 13.2941V10.4117H9.94119V12.0588L7.05884 13.2941Z"
      stroke={cssVar('currentTextColor')}
      strokeWidth="1.23529"
      strokeLinecap="round"
      strokeLinejoin="round"
    />
    <path
      d="M5 5.05884H12"
      stroke={cssVar('currentTextColor')}
      strokeWidth="1.23529"
      strokeLinecap="round"
      strokeLinejoin="round"
    />
  </svg>
);
