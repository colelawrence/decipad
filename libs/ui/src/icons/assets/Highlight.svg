<svg width="16" height="16" viewBox="0 0 16 16" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M5 3V5.05882C5.68627 6.29412 6.64706 9.09412 6.64706 10.4118H10.3529C10.3529 9.17647 12 5.05882 12 5.05882V3" stroke="black" stroke-width="1.23529" stroke-linecap="round" stroke-linejoin="round"/>
<path d="M7.05859 13.2942V10.4119H9.94095V12.0589L7.05859 13.2942Z" stroke="black" stroke-width="1.23529" stroke-linecap="round" stroke-linejoin="round"/>
<path d="M5 5.05884H12" stroke="black" stroke-width="1.23529" stroke-linecap="round" stroke-linejoin="round"/>
</svg>
