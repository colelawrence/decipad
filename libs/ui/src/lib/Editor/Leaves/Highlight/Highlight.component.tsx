import { Text } from '@chakra-ui/react';
import React from 'react';
import { RenderLeafProps } from 'slate-react';

export const Highlight = ({
  attributes,
  children,
}: RenderLeafProps): JSX.Element => {
  return (
    <Text as="span" bg="purple.100" p="1" {...attributes}>
      {children}
    </Text>
  );
};
