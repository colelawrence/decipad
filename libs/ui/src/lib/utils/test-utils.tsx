import { ChakraProvider } from '@chakra-ui/react';
import { render, RenderOptions } from '@testing-library/react';
import React from 'react';

import { Type } from '@decipad/language';
import { theme, ResultsContextProvider } from '../..';

const results = {
  test: {
    type: Type.build({
      type: 'number',
      unit: {
        unit: 'apples',
      },
    } as any),
    value: [20],
    errors: [],
  },
  noResult: null,
};

const AllTheProviders: React.FC = ({ children }) => {
  return (
    <ChakraProvider resetCSS theme={theme}>
      <ResultsContextProvider value={results}>
        {children}
      </ResultsContextProvider>
    </ChakraProvider>
  );
};

const customRender = (
  ui: JSX.Element,
  options?: Omit<RenderOptions, 'queries'>
) => render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from '@testing-library/react';
// override render method
export { customRender as render };
