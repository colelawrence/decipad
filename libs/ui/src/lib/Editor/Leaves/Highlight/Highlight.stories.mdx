import { Meta, Story, Canvas, Props } from '@storybook/addon-docs/blocks';
import { Highlight } from './Highlight.component';

<Meta
  title='Editor/Leaves/Highlight'
  parameters={{
    docs: {
      source: {
        code: `<Highlight
  attributes={{ 'data-slate-leaf': true }}
  leaf={{ text: 'Highlight text' }}
  text={{ text: 'Highlight text' }}
>
  Highlight text
</Highlight>`,
      },
    },
  }}
  argTypes={{
    children: {
      defaultValue: 'Hello World',
      description: 'The text content of the component',
      type: 'string',
    },
  }}
  component={Highlight}
/>

# Highlight

Highlight text by adding a background.

export const Template = (args) => <Highlight {...args} />;

<Canvas>
  <Story name='Highlight'>{Template.bind({})}</Story>
</Canvas>

<Props story='Highlight' />

