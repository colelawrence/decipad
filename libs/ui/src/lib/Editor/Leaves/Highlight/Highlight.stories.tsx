import React from 'react';
import { Highlight } from './Highlight.component';

export default {
  title: 'Editor/Leaves/Highlight',
};

export const B = () => (
  <Highlight
    text={{ text: 'hello world' }}
    leaf={{ text: 'hello world' }}
    attributes={{
      'data-slate-leaf': true,
    }}
  >
    The quick brown fox jumps over the lazy dog
  </Highlight>
);

B.storyName = 'Highlight';
