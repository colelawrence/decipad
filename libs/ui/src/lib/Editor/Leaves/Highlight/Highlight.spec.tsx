import { render } from 'test-utils';
import { Highlight } from './Highlight.component';

describe('Highlight Leaf', () => {
  it('renders', () => {
    const { container } = render(
      <Highlight
        attributes={{ 'data-slate-leaf': true }}
        leaf={{ text: 'Highlight text' }}
        text={{ text: 'Highlight text' }}
      >
        Highlight text
      </Highlight>
    );

    expect(container).toBeDefined();
  });

  it('matches snapshot', () => {
    const { container } = render(
      <Highlight
        attributes={{ 'data-slate-leaf': true }}
        leaf={{ text: 'Highlight text' }}
        text={{ text: 'Highlight text' }}
      >
        Highlight text
      </Highlight>
    );

    expect(container).toMatchSnapshot();
  });

  it('renders the string', () => {
    const { getByText } = render(
      <Highlight
        attributes={{ 'data-slate-leaf': true }}
        leaf={{ text: 'Highlight text' }}
        text={{ text: 'Highlight text' }}
      >
        Highlight text
      </Highlight>
    );

    getByText('Highlight text');
  });
});
