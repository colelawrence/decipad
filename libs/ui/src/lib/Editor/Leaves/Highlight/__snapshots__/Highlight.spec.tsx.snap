// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Highlight Leaf matches snapshot 1`] = `
.emotion-0 {
  background: var(--chakra-colors-purple-100);
  padding: var(--chakra-space-1);
}

<div>
  <span
    class="chakra-text emotion-0"
    data-slate-leaf="true"
  >
    Highlight text
  </span>
</div>
`;
